File: backend/internal/domain/bloodsearch/events/donation_confirmed.go
package events

import "time"

type DonorInfo struct {
	UserName         string `json:"userName"`
	PetName          string `json:"petName"`
	ProviderMaxID    string `json:"providerMaxId"`
	ProviderTelegram string `json:"providerTelegram"`
	Phone            string `json:"phone"`
	BloodGroup       string `json:"bloodGroup"`
}

type RecipientInfo struct {
	PetName    string `json:"petName"`
	BloodGroup string `json:"bloodGroup"`
}

type DonationConfirmed struct {
	DonorData     DonorInfo     `json:"donorData"`
	RecipientData RecipientInfo `json:"recipientData"`
	Volume        float64       `json:"volume"`
	CreatedAt     time.Time     `json:"createdAt"`
}

File: max-bot/src/events/recipient/donationConfirmed.ts
import { pinologger } from "../../instances";
import { sendMessageToUser } from "../../max";
import { getAppOpenKeyboard } from "../../keyboards";

import { generateDonationMessage } from "./helpers";

// Уведомление о подтвержденной донации (от реципиента донору).
interface DonationConfirmedEvent {
  donorData: {
    userName: string;
    petName: string;
    providerMaxId: string;
    providerTelegram: string;
    phone: string;
    bloodGroup: string;
  };
  recipientData: {
    petName: string;
    bloodGroup: string;
  };
  volume: number;
}

export const handleDonationConfirmed = async (
  event: DonationConfirmedEvent,
) => {
  const { donorData, recipientData, volume } = event;

  const targetId = donorData.providerMaxId;

  if (!targetId || targetId.trim() === "") {
    pinologger.warn(
      { donorUserName: donorData.userName },
      "Donor providerMaxId is empty, skipping notification",
    );
    return;
  }

  try {
    const message = generateDonationMessage({
      volume,
      recipientPetName: recipientData.petName,
      recipientBloodGroup: recipientData.bloodGroup,
    });

    await sendMessageToUser(targetId, message, {
      attachments: [getAppOpenKeyboard()],
    });

    pinologger.info(
      {
        targetId,
        volume,
      },
      "Sent donation confirmed notification to donor",
    );
  } catch (err) {
    pinologger.error(
      { error: err, targetId },
      "Failed to send donation confirmed notification",
    );
  }
};
